### PR TITLE
fix(capi): fix items mapping to match superdesk archive

### PR DESCRIPTION
**Pull request by petrjasek**
in elastic 2+ fields mapping must match across types

fix #912

### CODE REVIEW

**Comment by naspeh** [comment]
Looks good, but error is still here:
```
ERROR:elastic:mapping error, updating settings resource=items
Traceback (most recent call last):
  File "/opt/superdesk/env/lib/python3.5/site-packages/eve_elastic/elastic.py", line 302, in _put_resource_mapping
    es.indices.put_mapping(**kwargs)
  File "/opt/superdesk/env/lib/python3.5/site-packages/newrelic-2.66.0.49/newrelic/hooks/datastore_elasticsearch.py", line 50, in _nr_wrapper_Elasticsearch_method_
    return wrapped(*args, **kwargs)
  File "/opt/superdesk/env/lib/python3.5/site-packages/elasticsearch/client/utils.py", line 69, in _wrapped
    return func(*args, params=params, **kwargs)
  File "/opt/superdesk/env/lib/python3.5/site-packages/elasticsearch/client/indices.py", line 283, in put_mapping
    '_mapping', doc_type), params=params, body=body)
  File "/opt/superdesk/env/lib/python3.5/site-packages/elasticsearch/transport.py", line 327, in perform_request
    status, headers, data = connection.perform_request(method, url, params, body, ignore=ignore, timeout=timeout)
  File "/opt/superdesk/env/lib/python3.5/site-packages/elasticsearch/connection/http_urllib3.py", line 110, in perform_request
    self._raise_error(response.status, raw_data)
  File "/opt/superdesk/env/lib/python3.5/site-packages/elasticsearch/connection/base.py", line 114, in _raise_error
    raise HTTP_EXCEPTIONS.get(status_code, TransportError)(status_code, error_message, additional_info)
elasticsearch.exceptions.RequestError: TransportError(400, 'mapper_parsing_exception', 'analyzer [phrase_prefix_analyzer] not found for field [phrase]')
```
From log: https://test.superdesk.org/logs/misc/sdspr-916/index_from_mongo--elastic2.log

File: content_api/items/resource.py [hunk 1]
```diff
@@ -33,33 +24,33 @@
...
-    'slugline': {'type': 'string'},
-    'source': {'type': 'string'},
-    'subject': {'type': 'list', 'mapping': cv_mapping},
-    'keywords': {'type': 'list'},
-    'type': {'type': 'string'},
+    'slugline': metadata_schema['slugline'],
+    'source': metadata_schema['source'],
+    'subject': metadata_schema['subject'],
+    'keywords': metadata_schema['keywords'],
+    'type': metadata_schema['keywords'],
```
Comment (akintolga) [inline content_api/items/resource.py]: Would this be `metadata_schema['type']`?

File: content_api/items/resource.py [hunk 2]
```diff
@@ -33,33 +24,33 @@
...
     'urgency': {'type': 'integer'},
     'priority': {'type': 'integer'},
-    'uri': {'type': 'string'},
+    'uri': metadata_schema['guid'],
```
Comment (akintolga) [inline content_api/items/resource.py]: And this looks like a mismatch too.

**Comment by mdhaman** [comment]
@petrjasek there is `editor_state` in the schema mapping and it is dynamic.  We don't need to index editor field. Maybe mapping could be modified as below:

```        
'mapping': {
            'type': 'object',
            'include_in_all': False,
            'enabled': False
}
```

File: content_api/items/resource.py (hunk 1 above)
Comment (petrjasek) [inline content_api/items/resource.py]: ups

File: content_api/items/resource.py (hunk 2 above)
Comment (petrjasek) [inline content_api/items/resource.py]: here it's intentional, will add a comment there

**Comment by petrjasek** [comment]
@mdhaman seems like with `enabled: false` this `include_in_all` has no effect, so will avoid it to keep it simpler

**Pull request merged**